core/objlist_store.hpp

```diff
@@ -24,41 +24,63 @@
 
 namespace husky {
 
-typedef std::unordered_map<size_t, ObjListBase*> ObjListMap;
+typedef std::unordered_map<size_t, ObjListBase*> ObjListIdMap;
+typedef std::unordered_map<std::string, ObjListBase*> ObjListNameMap;
 
 class ObjListStore {
    public:
     template <typename ObjT>
     static ObjList<ObjT>& create_objlist() {
-        ObjListMap& objlist_map = get_objlist_map();
+        ObjListIdMap& objlist_map = get_objlist_map();
         auto* objlist = new ObjList<ObjT>();
         objlist_map.insert({objlist->get_id(), objlist});
         return *objlist;
     }
 
     template <typename ObjT>
+    static ObjList<ObjT>& create_objlist(const std::string& list_name) {
+        ObjListNameMap& objlist_name_map = get_objlist_name_map();
+        if (has_objlist(list_name)) {
+            throw base::HuskyException("A ObjList named "+list_name+" already exists.");
+        }
+        auto& objlist = create_objlist<ObjT>();
+        objlist_name_map.insert({list_name, &objlist});
+        return objlist;
+    }
+
+    template <typename ObjT>
     static ObjList<ObjT>& get_objlist(size_t id) {
-        ObjListMap& objlist_map = get_objlist_map();
-        auto* objlist = objlist_map[id];
+        auto* objlist = get_objlist_id_map().at(id);
+        return *static_cast<ObjList<ObjT>*>(objlist);
+    }
+
+    template <typename ObjT>
+    static ObjList<ObjT>& get_objlist(const std::string& list_name) {
+        auto* objlist = get_objlist_name_map().at(list_name);
         return *static_cast<ObjList<ObjT>*>(objlist);
     }
 
     static bool has_objlist(size_t id);
+    static bool has_objlist(const std::string& list_name);
     static void drop_objlist(size_t id);
+    static void drop_objlist(const std::string& list_name);
     static void drop_all_objlists();
 
     static void init_objlist_map();
     static void free_objlist_map();
-    static ObjListMap& get_objlist_map();
+    static ObjListIdMap& get_objlist_map();
+    static ObjListIdMap& get_objlist_id_map();
+    static ObjListNameMap& get_objlist_name_map();
 
     static inline size_t size() {
-        if (s_objlist_map == nullptr)
+        if (s_objlist_id_map == nullptr)
             return 0;
-        return s_objlist_map->size();
+        return s_objlist_id_map->size();
```

Comment:
That's why I just added some comments to avoid confusion...

I want the `id_map` to be the primary index, and the `name_map` as the auxiliary/non-compulsory index.